### DISCUSSION TITLE
I don't think we should do a parseEther in useScaffoldContractWrite for the value

### DISCUSSION THREAD

**Discussion by austintgriffith** [scaffold-eth/scaffold-eth-2#516] [General]
i don't think we should do parseEther here https://github.com/scaffold-eth/scaffold-eth-2/blob/main/packages/nextjs/hooks/scaffold-eth/useScaffoldContractWrite.ts#L44C20-L44C30

I think people should pass in bigints or whatever is coming back from useScaffoldContractRead for uint256s

**Comment by damianmarti** [comment]
Maybe we can allow string or BigNumber for value there...

**Comment by carletex** [comment]
Initially, we opted for parsing ETH internally since we thought it'd improve DX. But in terms of having a good mental model, maybe it's better for newcomers to understand that a value will always need to be expressed in wei.

So passing bigints could be a good option.

**Comment by carletex** [comment]
Fixed on https://github.com/scaffold-eth/scaffold-eth-2/pull/548
